airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -0,0 +1,155 @@
+                    raise Exception("Google Cloud Dataflow job {} was cancelled.".format(
+                        self._job['name']))
+                elif 'JOB_STATE_RUNNING' == self._job['currentState']:
+                    time.sleep(10)
+                else:
+                    logging.debug(str(self._job))
+                    raise Exception(
+                        "Google Cloud Dataflow job {} was unknown state: {}".format(
+                            self._job['name'], self._job['currentState']))
+            else:
+                time.sleep(15)
+
+            self._job = self._get_job()
+
+    def get(self):
+        return self._job
+
+
+class _DataflowJava(object):
+    def __init__(self, cmd):
+        self._proc = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE,
+                                      stderr=subprocess.PIPE)
+
+    def _line(self, fd):
+        if fd == self._proc.stderr.fileno():
+            line = self._proc.stderr.readline()
+            return line
+        if fd == self._proc.stdout.fileno():
+            line = self._proc.stdout.readline()
+            return line
+
+    @staticmethod
+    def _extract_job(line):
+        if line is not None:
+            if line.startswith("Submitted job: "):
+                return line[15:-1]
+
+    def wait_for_done(self):
+        reads = [self._proc.stderr.fileno(), self._proc.stdout.fileno()]
+        logging.info("Start waiting for DataFlow process to complete.")
+        while self._proc.poll() is None:
+            ret = select.select(reads, [], [], 5)
+            if ret is not None:
+                for fd in ret[0]:
+                    line = self._line(fd)
+                    logging.debug(line[:-1])
+            else:
+                logging.info("Waiting for DataFlow process to complete.")
+        if self._proc.returncode is not 0:
+            raise Exception("DataFlow jar failed with return code {}".format(
+                self._proc.returncode))
+
+
+class DataFlowHook(GoogleCloudBaseHook):
+    def __init__(self,
+                 gcp_conn_id='google_cloud_default',
+                 delegate_to=None):
+        """
+        :param scope: The scope of the hook (read only, read write, etc). See:
+            https://cloud.google.com/storage/docs/authentication?hl=en#oauth-scopes
+        :type scope: string
+        """
+        super(DataFlowHook, self).__init__(gcp_conn_id, delegate_to)
+
+    def get_conn(self):
+        """
+        Returns a Google Cloud Storage service object.
+        """
+        http_authorized = self._authorize()
+        return build('dataflow', 'v1b3', http=http_authorized)
+
+    def start_java_dataflow(self, task_id, variables, dataflow):
+        name = task_id + "-" + str(uuid.uuid1())[:8]
+        cmd = self._build_cmd(task_id, variables, dataflow, name)
+        _DataflowJava(cmd).wait_for_done()
+        _DataflowJob(self.get_conn(), "vex-eu-cloud-sql-001", name).wait_for_done()
```

Comment:
The second argument should be `variables['project']` instead of `"vex-eu-cloud-sql-001"`